StanfordHCI/foundry
# Edit Team

The author should be able to edit members, events, and interactions. If I have missed a feature, let me know.

Notes:
- Added pauseTeam.js file.
- Added flashTeamsJSON["paused"]. If this is true and in_progress is true, it means that the team is being edited. 
- Changed all updateStatus(true/false) to updateStatus(). I just kept 3 calls in awareness.js. Those functions calls are for the time that we start or end the team.
- From now on, we shouldn't call don't call updateStatus(false/true) unless we are sure the team is not started/started when the updateStatus function is being called.

Minor note: 
I have used "paused" in the code instead of "edit" because the button was first called paused and then we decided to change it to edit. 

How to test:
- First create a team with members and handoffs and collaborations and make sure everything works fine. 
- Create the three different roles (worker, PC, and client) and invite them to Foundry.
- Start the team
- Only the author should see the edit team button. Only the author can edit the team.
- Have a running task while testing. Save the team and press edit again while you are testing. You want to make sure that when you save the team it doesn't revert back to edit mode if there are in progress or delayed tasks.
- Go to edit mode.
- Draw an event.
- Gdrive folder should be added for the event. 
- Edit the task. Make sure other views can see the change in the task.
- Delete a task (Gdrive folder is not deleted because there might be files inside a folder and we don't want to lose them).
- Drag events.
- The handoff button, collaboration button, and the drag handles should show up in edit mode only for the author but not other users. refresh the pages and check again.
- Draw, delete handoffs and collaborations.
- You shouldn't be able to edit completed or in progress tasks
- Edit members: create a new role, add them to events, delete the role. Make sure other views can see the changes.
- Right now, even the worker can edit the project overview. I didn't change this.
